### PR TITLE
[browser] Fix SIMD+EH check

### PR DESCRIPTION
**Pull request by maraf**
- Before this change the `linkerWasmEnableSIMD` and `linkerWasmEnableEH` used in `configureRuntimeStartup` to check support for these features always had the default (`true`) value. Throwing assert error even when SIMD and EH was disabled for build.
- Call `cwraps.mono_wasm_abort` from `runtimeHelpers.abort` only after cwraps are ready (`onRuntimeInitializedAsync`).

### CODE REVIEW

File: src/mono/wasm/runtime/startup.ts [hunk 1]
```diff
@@ -363,11 +367,10 @@ function mono_wasm_pre_init_essential(isWorker: boolean): void {
     }
 
     init_c_exports();
-    runtimeHelpers.mono_wasm_exit = cwraps.mono_wasm_exit;
+    runtimeHelpers.mono_wasm_exit = () => {
+        throw new Error("Mono shutdown");
+    };
     runtimeHelpers.abort = (reason: any) => {
```
[the remark this answers is not in the document]
Comment (pavelsavara) [reply src/mono/wasm/runtime/startup.ts:373]: And perhaps those should be copied from `loaderHelper`

File: src/mono/wasm/runtime/startup.ts [hunk 2]
```diff
@@ -41,12 +41,7 @@ import { assertNoProxies } from "./gc-handles";
 const MONO_PTHREAD_POOL_SIZE = 4;
 
 export async function configureRuntimeStartup(): Promise<void> {
-    if (linkerWasmEnableSIMD) {
-        mono_assert(await loaderHelpers.simd(), "This browser/engine doesn't support WASM SIMD. Please use a modern version. See also https://aka.ms/dotnet-wasm-features");
-    }
-    if (linkerWasmEnableEH) {
-        mono_assert(await loaderHelpers.exceptions(), "This browser/engine doesn't support WASM exception handling. Please use a modern version. See also https://aka.ms/dotnet-wasm-features");
-    }
+
```
Comment (radical) [inline src/mono/wasm/runtime/startup.ts:44]:
```suggestion
```

**Comment by radical** [comment]
> Before this change the linkerWasmEnableSIMD and linkerWasmEnableEH used in configureRuntimeStartup to check support for these features always had the default (true) value. Throwing assert error even when SIMD and EH was disabled for build.

So, this always failed on browsers that didn't support SIMD/EH, even when SIMD was disabled in the build? Can we test this with chrome/v8?

File: src/mono/wasm/runtime/startup.ts [hunk 3]
```diff
@@ -470,6 +466,8 @@ async function instantiate_wasm_module(
 
         replace_linker_placeholders(imports);
         const assetToLoad = await loaderHelpers.wasmDownloadPromise.promise;
+
+        await ensureUsedWasmFeatures();
```
Comment (pavelsavara) [inline src/mono/wasm/runtime/startup.ts:470]: we could actually bomb even before downloading the wasm file, few lines earlier.

Comment (maraf) [reply src/mono/wasm/runtime/startup.ts]: I have fired it before awaiting wasm download, but awaited after. Makes sense?

**Comment by maraf** [comment]
> So, this always failed on browsers that didn't support SIMD/EH, even when SIMD was disabled in the build? 
Exactly. 

> Can we test this with chrome/v8?
If we install Chrome 85 or Firefox 88, we can test it. I can look into it in a separate PR.

**Review by pavelsavara — APPROVED** [review]

**Review by radical — APPROVED** [review]

**Comment by lewing** [comment]
closed in favor of https://github.com/dotnet/runtime/pull/92439

**Pull request merged**